Content.Server/GameObjects/Components/Access/IdCardConsoleComponent.cs

```diff
@@ -0,0 +1,202 @@
+using System.Collections.Generic;
+using System.Linq;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Server.Interfaces;
+using Content.Server.Interfaces.GameObjects;
+using Content.Shared.Access;
+using Content.Shared.GameObjects.Components.Access;
+using Robust.Server.GameObjects.Components.Container;
+using Robust.Server.GameObjects.Components.UserInterface;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Log;
+
+namespace Content.Server.GameObjects.Components.Access
+{
+    [RegisterComponent]
+    [ComponentReference(typeof(IActivate))]
+    public class IdCardConsoleComponent : SharedIdCardConsoleComponent, IActivate
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IServerNotifyManager _notifyManager;
+        [Dependency] private readonly ILocalizationManager _localizationManager;
+#pragma warning restore 649
+
+        private BoundUserInterface _userInterface;
+        private Container _privilegedIdContainer;
+        private Container _targetIdContainer;
+        private AccessReader _accessReader;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            _privilegedIdContainer = ContainerManagerComponent.Ensure<Container>($"{Name}-privilegedId", Owner);
```

Comment:
Does it make sense to make this use a `ContainerSlot`? It's intended to be used for single slots like this.

Reply:
I had no idea such thing existed. Yes, I'll make it use that.